### PR TITLE
(feat) Improved field validation UX

### PR DESCRIPTION
**Pull request by denniskigen**
# Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR improves the appearance and behavior of various fields when they fail validation. Specifically, fields with invalid values now have an error outline indicating that their content is invalid. Additionally, various validation error messages and their equivalent locale-specific translations have been amended to communicate their intent more effectively. In summary:

- Fields with invalid values now have a visible error outline, indicating that their content is invalid.
- I've updated validation error messages to improve their clarity and effectiveness.
- I've updated locale-specific translations for validation error messages.

These improvements enhance the user experience by providing clear and specific feedback when fields fail validation, making it easier for users to understand and address validation issues.

## Screenshots

> Before (note the absence of styling around inputs that fail validation)

https://github.com/openmrs/openmrs-ngx-formentry/assets/8509731/120247c9-1523-4e32-b8e6-850adb41314b

>  After

https://github.com/openmrs/openmrs-ngx-formentry/assets/8509731/aa94a2b3-cd1b-4bb6-bc13-754246de28ce

## Related Issue

*None*

## Other

*None*

### CODE REVIEW

File: projects/ngx-formentry/src/components/number-input/number-input.component.html [hunk 1]
```diff
@@ -29,8 +30,22 @@
       [attr.step]="step"
       [disabled]="disabled"
       [required]="required"
+      [attr.data-invalid]="invalid ? invalid : null"
+      [placeholder]="placeholder"
       (input)="onNumberInputChange($event)"
     />
+    <svg
```
Comment (denniskigen) [inline projects/ngx-formentry/src/components/number-input/number-input.component.html:37]: I've updated the `NumberInput` component API to match the related [Carbon component](https://angular.carbondesignsystem.com/?path=/docs/components-number-input--docs) where the code was originally obtained from.

File: projects/ngx-formentry/src/form-entry/form-renderer/form-renderer.component.html [hunk 2]
```diff
@@ -289,8 +298,8 @@
             [minlength]="node.question.extras.questionOptions.minLength"
             [formControlName]="node.question.key"
             [attr.placeholder]="node.question.placeholder | translate"
-          >
-          </ofe-number-input>
+            [invalid]="node.control.touched && !node.control.valid"
```
Comment (denniskigen) [inline projects/ngx-formentry/src/form-entry/form-renderer/form-renderer.component.html:301]: I'm basing the invalid validation state on whether the control has been touched and its value is invalid.

File: projects/ngx-formentry/src/form-entry/form-renderer/form-renderer.component.scss [hunk 3]
```diff
@@ -183,3 +183,19 @@ ng-select.form-control {
...
+    display: flex;
+    align-items: center;
+
+    svg {
+      margin-left: 1rem;
+      margin-block-start: 0;
+    }
+  }
+
+  .cds--inline-notification--low-contrast::before {
```
Comment (denniskigen) [inline projects/ngx-formentry/src/form-entry/form-renderer/form-renderer.component.scss:198]: Here I've just removed the bottom border of the error container in the error pane that gets rendered after attempting to submit a form without filling all the mandatory fields.

File: src/translations/km.json [hunk 4]
```diff
@@ -1,67 +1,66 @@
...
+  "requiredField": "ប្រអប់​នេះ​ត្រូវតែ​បំពេញ",
+  "saveAndCloseButton": "រក្សាទុក និងបិទ",
+  "selectWeeks": "ជ្រើសរើសសប្តាហ៍",
+  "set": "កំណត់",
+  "submitting": "បញ្ជូន",
+  "to": "ទៅ",
+  "tryOpeningAnotherForm": "ព្យាយាមបើកទម្រង់ផ្សេងទៀត",
+  "useValue": "ប្រើតម្លៃ",
+  "weeks": "សប្តាហ៍",
+  "yearsAgo": " ឆ្នាំមុន"
```
Comment (denniskigen) [inline src/translations/km.json:65]: I've arranged all these translation keys and strings in ascending order for readability.

**Comment by donaldkibet** [comment]
@denniskigen LGTM

**Review by donaldkibet — APPROVED** [review]

File: projects/ngx-formentry/src/components/ngx-remote-select/ngx-remote-select.component.ts [hunk 5]
```diff
@@ -40,11 +41,12 @@ export class RemoteSelectComponent implements OnInit, ControlValueAccessor {
   value = [];
   loading = false;
   searchText = '';
-  notFoundMsg = 'match no found';
+  notFoundMsg = 'Match not found';
```
[the remark this answers is not in the document]
Comment (denniskigen) [reply projects/ngx-formentry/src/components/ngx-remote-select/ngx-remote-select.component.ts:44]: Is translating this simply a matter of calling `this.translate.instant` and passing in a key?

[the remark this answers is not in the document]
Comment (ibacher) [reply projects/ngx-formentry/src/components/ngx-remote-select/ngx-remote-select.component.ts]: I believe so, yes.

**Comment by denniskigen** [comment]
Thanks for the reviews!

**Pull request merged**